### PR TITLE
Shmem region: support huge pages via path to hugetlbfs mount

### PR DESCRIPTION
**Pull request by rbx**
- Support huge pages via path to hugetlbfs mount and/or flags to mmap. Adds optional path/flags arguments to interface of region creation.
- Store region info in shared memory with paths and flags,
- Shmem refactoring: move device counter and monitor launcher from factory to manager class,
- Formatting...

---

Checklist:

* [x] Rebased against `dev` branch
* [x] My name is in the resp. CONTRIBUTORS/AUTHORS file
* [x] Followed [the seven rules of great commit messages](https://chris.beams.io/posts/git-commit/#seven-rules)

### CODE REVIEW

File: fairmq/shmem/runMonitor.cxx [hunk 1]
```diff
@@ -93,26 +87,22 @@ int main(int argc, char** argv)
...
+        if (vm.count("help")) {
             cout << "FairMQ Shared Memory Monitor" << endl << desc << endl;
             return 0;
         }
 
         notify(vm);
 
-        if (runAsDaemon)
-        {
+        if (runAsDaemon) {
```
[the remark this answers is not in the document]
Comment (dennisklein) [reply fairmq/shmem/runMonitor.cxx]: false positive

[the remark this answers is not in the document]
Comment (rbx) [reply fairmq/shmem/runMonitor.cxx]: false positive.

File: fairmq/shmem/runMonitor.cxx [hunk 2]
```diff
@@ -93,26 +87,22 @@ int main(int argc, char** argv)
...
 
-        if (shmId == "")
-        {
+        if (shmId == "") {
             shmId = buildShmIdFromSessionIdAndUserId(sessionName);
         }
 
-        if (cleanup)
-        {
+        if (cleanup) {
```
[the remark this answers is not in the document]
Comment (dennisklein) [reply fairmq/shmem/runMonitor.cxx]: false positive

[the remark this answers is not in the document]
Comment (rbx) [reply fairmq/shmem/runMonitor.cxx]: false positive.

**Pull request merged**